GymBuddies/ViewControllers/RegisterViewController.m

```diff
@@ -0,0 +1,122 @@
+//
+//  RegisterViewController.m
+//  GymBuddies
+//
+//  Created by Gael Rodriguez Gomez on 7/7/22.
+//
+
+#import "RegisterViewController.h"
+#import "ParseAPIManager.h"
+#import "DataModelBlocks.h"
+#import "CommonValidations.h"
+
+@interface RegisterViewController ()
+@property (weak, nonatomic) IBOutlet UITextField *emailField;
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+@property (weak, nonatomic) IBOutlet UITextField *confirmPasswordField;
+@property (weak, nonatomic) IBOutlet UISegmentedControl *trainingLevelControl;
+@property (weak, nonatomic) IBOutlet UISegmentedControl *workoutPlaceControl;
+
+@end
+
+@implementation RegisterViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+}
+
+
+#pragma mark - Buttons interactions
+
+- (IBAction)didTapSubmit:(id)sender {
+    
+    NSString *email = [CommonValidations standardizeUserAuthInput:self.emailField.text];
+    NSString *username = [CommonValidations standardizeUserAuthInput:self.usernameField.text];
+    NSString *password = [CommonValidations standardizeUserAuthInput:self.passwordField.text];
+    NSString *confirmPassword = [CommonValidations standardizeUserAuthInput:self.confirmPasswordField.text];
+    
+    if([self _lookForEmptyFields:username email:email password:password confirmPassword:confirmPassword]){
+        [self _emptyFieldAlert];
+        return;
+    }
+    
+    if(![self _checkPasswordsMatching]){
+        [self _passwordsNotMatchingAlert];
+        return;
+    }
+    
+    TrainingLevels level = [self.trainingLevelControl selectedSegmentIndex];
+    
+    WorkoutPlace place = [self.workoutPlaceControl selectedSegmentIndex];
+    
+    PFUser *user = [[PFUser alloc]init];
+    user[@"email"] = email;
+    user[@"username"] = username;
+    user[@"password"] = password;
+    user[@"trainingLevel"] = [NSNumber numberWithLong:level];
+    user[@"workoutPlace"] = [NSNumber numberWithLong:place];
+    
+    [ParseAPIManager signUp:user completion:^(BOOL succeeded, NSError * _Nonnull error) {
+        if(error != nil){
+            [self _submitErrorAlert:[error localizedDescription]];
+        } else {
+            [self.navigationController popViewControllerAnimated:YES];
+        }
+    }];
+    
+}
+
+#pragma mark - Validations
+
+    
+- (BOOL)_lookForEmptyFields:(NSString *)username
+                      email:(NSString *)email
+                   password:(NSString *)password
+            confirmPassword:(NSString *)confirmPassword {
+    
+    return (username.length == 0 || password.length == 0 || confirmPassword == 0 || email.length == 0);
+
+}
+
```

Comment:
not a requirement but you can consider just making a helper that takes in a list of strings and returns true if ANY are empty